### PR TITLE
Allow multiple order parameters to contribute to interface stress

### LINKED ISSUE #11006: Allow multiple order parameters to influence interface stress
## Rationale

Enables polycrystalline and multiphase simulations with interface stress.

## Description

Make the material accept vectors of parameters rather than a single set of parameters.

## Impact

Added capability

### PR DESCRIPTION
**Pull request by dschwen**
Added capability and a test for it.

Closes #11006

### CODE REVIEW

**Comment by dschwen** [comment]
What happened in the pre-check?!

```
`git diff-files --name-status ` returned 128
fatal: ambiguous argument '': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
ERROR: exiting with code 2
```

**Comment by brianmoose** [comment]
You will need to rebase to the latest for this to work.
This was originally caused by a recent PR that we had to force through because clang format always gave the above error.

File: modules/tensor_mechanics/src/materials/ComputeInterfaceStress.C
Comment (laagesen) [inline modules/tensor_mechanics/src/materials/ComputeInterfaceStress.C]: defines -> define

File: modules/tensor_mechanics/src/materials/ComputeInterfaceStress.C
Comment (laagesen) [inline modules/tensor_mechanics/src/materials/ComputeInterfaceStress.C]: Since you are allowing either a single stress or a different stress for each OP (as in the error check below) I think there should be some mention of this in the help text here

**Review by laagesen — CHANGES_REQUESTED** [review]
Just a few small things, looks good otherwise.

File: modules/tensor_mechanics/src/materials/ComputeInterfaceStress.C
Comment (amjokisaari) [reply modules/tensor_mechanics/src/materials/ComputeInterfaceStress.C]: Seconding Larry's comment here.

File: modules/tensor_mechanics/src/materials/ComputeInterfaceStress.C
Comment (amjokisaari) [inline modules/tensor_mechanics/src/materials/ComputeInterfaceStress.C]: This part is also unclear in the parameter description.  This parameter is now a vector as well, so you can have multiple ranges?

File: modules/tensor_mechanics/src/materials/ComputeInterfaceStress.C
Comment (amjokisaari) [inline modules/tensor_mechanics/src/materials/ComputeInterfaceStress.C]: Strong preference: error messages tell me which input block / MOOSE class is failing.

**Comment by dschwen** [comment]
Documentation test seems to be broken @brianmoose

File: modules/tensor_mechanics/src/materials/ComputeInterfaceStress.C
Comment (amjokisaari) [inline modules/tensor_mechanics/src/materials/ComputeInterfaceStress.C]: What's the purpose of this if statement and continue statement?

File: modules/tensor_mechanics/src/materials/ComputeInterfaceStress.C
Comment (amjokisaari) [inline modules/tensor_mechanics/src/materials/ComputeInterfaceStress.C]: Is there documentation for this stress calculation?

**Review by amjokisaari — CHANGES_REQUESTED** [review]

File: modules/tensor_mechanics/src/materials/ComputeInterfaceStress.C
Comment (dschwen) [reply modules/tensor_mechanics/src/materials/ComputeInterfaceStress.C]: "no interface, return zero stress" :-)

It means if the norm of the order parameter gradient is zero (within numerical tolerance) there is no interface to contribute to an interfacial planar stress

**Comment by brianmoose** [comment]
@aeslaughter is trying to get his new documentation system in. It will be broken until his PR is merged. Please do not merge anything until then. I will invalidate all the failing documentation jobs when it is in.

File: modules/tensor_mechanics/src/materials/ComputeInterfaceStress.C
Comment (dschwen) [reply modules/tensor_mechanics/src/materials/ComputeInterfaceStress.C]: moose/docs/content/documentation/systems/Materials/tensor_mechanics/ComputeInterfaceStress.md

File: modules/tensor_mechanics/src/materials/ComputeInterfaceStress.C
Comment (dschwen) [reply modules/tensor_mechanics/src/materials/ComputeInterfaceStress.C]: Well, you are in for a treat, thanks to @rwcarlsen's work on the parser. Commit coming up...

**Comment by dschwen** [comment]
Using `mooseError`:
```
*** ERROR ***
Supply either one single stress or one per order parameter
```

Using `paramError`:
```
*** ERROR ***
/Users/schwd/Programs/marmot-problems/pressurized_bubble/grains.i:190: (Materials/interfacestress/stress) Supply either one single stress or one per order parameter
```

File: modules/tensor_mechanics/src/materials/ComputeInterfaceStress.C
Comment (amjokisaari) [reply modules/tensor_mechanics/src/materials/ComputeInterfaceStress.C]: I get that, but the lines themselves are extraneous, nothing is being done...

File: modules/tensor_mechanics/src/materials/ComputeInterfaceStress.C
Comment (amjokisaari) [inline modules/tensor_mechanics/src/materials/ComputeInterfaceStress.C]: Much better.

File: modules/tensor_mechanics/src/materials/ComputeInterfaceStress.C
Comment (dschwen) [reply modules/tensor_mechanics/src/materials/ComputeInterfaceStress.C]: That is incorrect

Comment (andrsd) [reply modules/tensor_mechanics/src/materials/ComputeInterfaceStress.C]: @amjokisaari If nothing was being done and you just removed those lines, then look what would happen on line 83:
```
const Real s = _stress[i] / std::sqrt(grad_norm_sq);
```

But, I would agree that the word "return" in the comment is misleading...

Comment (amjokisaari) [reply modules/tensor_mechanics/src/materials/ComputeInterfaceStress.C]: Ok, so I had to look up continue statements and I stand corrected.  Still don't like the construction of hiding the condition in an if and then breaking out of the loop, but I'm not going to bother considering if there's an alternative way to construct this.

**Comment by dschwen** [comment]
It is not breaking out of the loop and it is preferred to ever increasing
indentation.

File: modules/tensor_mechanics/src/materials/ComputeInterfaceStress.C
Comment (dschwen) [reply modules/tensor_mechanics/src/materials/ComputeInterfaceStress.C]: I hope the updated comment clarifies this

File: libmesh
Comment (dschwen) [reply libmesh]: nargh!!

Comment (dschwen) [reply libmesh]: can't we put `libmesh` into `.gitignore` so that it is only added with `git add -f` ?

File: modules/tensor_mechanics/src/materials/ComputeInterfaceStress.C
Comment (amjokisaari) [reply modules/tensor_mechanics/src/materials/ComputeInterfaceStress.C]: okay.  The comment is better, I see what's going on now.

File: modules/tensor_mechanics/src/materials/ComputeInterfaceStress.C
Comment (dschwen) [reply modules/tensor_mechanics/src/materials/ComputeInterfaceStress.C]: I had to look this up myself but `paramname` is substituted with each vector entry (i.e. teh check applies to all entries), `paramname_size` checks the size of the vector, and `paramname_3` checks `vector[3]` etc.

Turns out I wrote the range checking for vector parameters myself 3.5 years ago :-D

**Review by amjokisaari — APPROVED** [review]

**Comment by dschwen** [comment]
@aeslaughter here I'm getting 

```
MooseDocs/test/common.class_database: IOError: [Errno 2] No such file or directory: '/opt/civet/build_0/moose/test/tests/test_harness/job_unittesting_pbs/good.i'
```

That file does indeed not exist in the repo. What's up with that.

**Comment by dschwen** [comment]
Halo. I cannot reproduce this and have no idea where this failure comes from.

**Review by laagesen — APPROVED** [review]

**Review by andrsd — APPROVED** [review]

**Pull request merged**
